codex-rs/mcp-server/src/codex_message_processor.rs

```diff
@@ -212,6 +229,20 @@ impl CodexMessageProcessor {
                         })
                         .await;
 
+                    // Send an auth status change notification.
+                    if success {
+                        let notification = AuthStatusChangeNotification {
+                            auth_method: Some(AuthMethod::ChatGPT),
+                        };
+                        let params = serde_json::to_value(&notification).ok();
+                        outgoing_clone
+                            .send_notification(OutgoingNotification {
```

Comment:
With my recommended change to introduce `ServerNotification`, we should update `send_notification` to take `ServerNotification`, which will make this simpler.